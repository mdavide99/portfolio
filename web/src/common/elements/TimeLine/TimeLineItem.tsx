import { ReactNode } from "react";

interface TimeLineItemProps {
  title: string;
  subTitle: string;
  period: string;
  children?: ReactNode;
}

function TimeLineItem(props: TimeLineItemProps) {
  return (
    <div className="lg:flex lg:justify-between lg:max-w-4xl mx-auto mb-8">
      <div className="mb-2 lg:w-52">
        <span className="font-light">{props.period}</span>
      </div>

      <div className="lg:w-3/4">
        <h3 className="text-xl leading-tight uppercase font-bold text-neutral-900 dark:text-white">
          {props.title}
        </h3>

        <h4 className="text-lg leading-tight font-medium my-2 md:my-1 text-neutral-600 dark:text-neutral-50">
          {props.subTitle}
        </h4>

        <div className="mb-4">{props.children}</div>
      </div>
    </div>
  );
}

export { TimeLineItem };
