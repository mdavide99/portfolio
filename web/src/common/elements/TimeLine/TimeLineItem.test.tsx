import { TimeLineItem } from "./TimeLineItem";
import { render } from "@testing-library/react";

describe("TimeLineItem", () => {
  it("renders TimeLineItem correctly", () => {
    const { container } = render(
      <TimeLineItem
        title={"Title"}
        subTitle={"SubTitle"}
        period={"from ... to ..."}
      >
        <p>Some Contents</p>
      </TimeLineItem>
    );
    expect(container).toMatchSnapshot();
  });
});
