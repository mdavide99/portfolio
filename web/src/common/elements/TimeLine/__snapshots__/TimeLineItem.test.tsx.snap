// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeLineItem renders TimeLineItem correctly 1`] = `
<div>
  <div
    class="lg:flex lg:justify-between lg:max-w-4xl mx-auto mb-8"
  >
    <div
      class="mb-2 lg:w-52"
    >
      <span
        class="font-light"
      >
        from ... to ...
      </span>
    </div>
    <div
      class="lg:w-3/4"
    >
      <h3
        class="text-xl leading-tight uppercase font-bold text-neutral-900 dark:text-white"
      >
        Title
      </h3>
      <h4
        class="text-lg leading-tight font-medium my-2 md:my-1 text-neutral-600 dark:text-neutral-50"
      >
        SubTitle
      </h4>
      <div
        class="mb-4"
      >
        <p>
          Some Contents
        </p>
      </div>
    </div>
  </div>
</div>
`;
